project1_tiny_calculator/lib/calculate_provider.dart

```diff
@@ -1,47 +1,54 @@
 import 'package:flutter/material.dart';
 
 class CalculateProvider extends ChangeNotifier {
-  List<String> operations = ['+', '/', '-', '*'];
+  final List<String> _operations = ['+', '/', '-', '*'];
   final TextEditingController firstNumberController = TextEditingController();
   final TextEditingController secondNumberController = TextEditingController();
   String result = '';
-  int currentIndex = 0;
+  int _currentIndex = 0;
 
-  String get currentOperation => operations[currentIndex];
+  String get currentOperation => _operations[_currentIndex];
 
   void changeOperator(isUp) {
-    currentIndex = (currentIndex + (isUp ? 1 : -1)) % operations.length;
-    if (currentIndex < 0) currentIndex += operations.length;
-
+    _currentIndex = (_currentIndex + (isUp ? 1 : -1)) % _operations.length;
+    if (_currentIndex < 0) _currentIndex += _operations.length;
     calculate();
-    notifyListeners();
+  }
+
+  void numberProvider() {
```

Comment:
totally doesn't look like a method name.
void methods should be in imperative tense, like doSomething()
For example, this one can be called addListeners() or listenToChanges() etc.

Reply:
second point: where are you calling this? you're not calling it anywhere, so this code is not executing at all. So you're not listening and hence nothing is happening